ui/appui-react/src/appui-react/widgets/Widget.tsx

```diff
@@ -47,4 +48,28 @@ export interface Widget {
   readonly label?: string | ConditionalStringValue;
   readonly priority?: number;
   readonly tooltip?: string | ConditionalStringValue;
+  /** Describes layout specific configurations of a widget.
+   * @alpha
+   */
+  readonly layouts?: WidgetLayouts;
+}
+
+/** Describes widget configuration specific for each layout.
+ * @alpha
+ */
+export interface WidgetLayouts {
+  /** Widget configuration in a panels layout. */
+  readonly standard?: PanelsLayoutWidget;
+  /** Widget configuration in a custom layout. */
+  readonly [layoutId: string]: Object | undefined;
```

Comment:
I think we could remove the key-value pair for now. Making `standard` an optional property in the future would be a breaking change...

Reply:
Alternatively, we could use a default location for widgets/toolbar items if `standard` is not provided. But then we will probably want to add a flag to the `frontstage` to disable that behavior in the future.

Reply:
Since it is a property like any other, making it optional is a breaking change.
I.e. in v1 (where property is required), a developer is accessing that w/o any checks. After we update our definitions it will either result in TS errors or runtime errors.
Also, this will not solve the original issue of forcing required location values, since we have to keep the `layouts` property  _optional_. TS will not complain about `standard` property unless `layouts` is set...